frontend/src/metabase/nav/components/AppBar/AppBarSmall.tsx

```diff
@@ -63,34 +67,41 @@ const AppBarSmall = ({
 
   return (
     <AppBarRoot>
-      <AppBarHeader isSubheaderVisible={isSubheaderVisible}>
-        <AppBarMainContainer>
-          <AppBarToggleContainer>
-            {isNavBarEnabled && (
-              <AppBarToggle
-                isNavBarOpen={isNavBarVisible}
-                onToggleClick={onToggleNavbar}
-              />
+      {isHeaderVisible && (
+        <AppBarHeader isSubheaderVisible={isSubheaderVisible}>
+          <AppBarMainContainer>
+            <AppBarToggleContainer>
+              {isNavBarEnabled && (
+                <AppBarToggle
+                  isSmallAppBar
+                  isNavBarOpen={isNavBarVisible}
+                  onToggleClick={onToggleNavbar}
```

Comment:
The rest of the the change that isn't commented should almost, if not, the same.